R/serializer.R

```diff
@@ -231,6 +231,17 @@ serializer_unboxed_json <- function(auto_unbox = TRUE, ..., type = "application/
   serializer_json(auto_unbox = auto_unbox, ..., type = type)
 }
 
+#' @describeIn serializers GeoJSON serializer. See also [geojsonsf::sf_geojson()] and [[geojsonsf::sfc_geojson()]].
+#' @export
+serializer_geojson <- function(..., type = "application/geo+json") {
+  serializer_content_type(type, function(val) {
+    if (inherits(val, "sfc")) return(geojsonsf::sfc_geojson(val, ...))
+    if (inherits(val, "sf"))  return(geojsonsf::sf_geojson(val, ...))
+    if (!requireNamespace("geojsonsf", quietly = TRUE)) {
+      stop("`geojsonsf` must be installed for `serializer_geojson` to work")
+    }
+  })
+}
```

Comment:
Let's move the `requireNamespace()` check up so that if the serializer is called, the error could be thrown immediately.

Let's also add a generic "this isn't a geojson object" error message if not an `sf` or `sfc` object. (Feel free to change the wording)

```suggestion
serializer_geojson <- function(..., type = "application/geo+json") {
  if (!requireNamespace("geojsonsf", quietly = TRUE)) {
    stop("`geojsonsf` must be installed for `serializer_geojson()` to work")
  }
  serializer_content_type(type, function(val) {
    if (inherits(val, "sfc")) return(geojsonsf::sfc_geojson(val, ...))
    if (inherits(val, "sf"))  return(geojsonsf::sf_geojson(val, ...))
    stop("Did not receive a `sf` or `sfc` object")
  })
}
```